lib/util/timeout.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+/**
+ * A {@link Timeout} represents a resettable and clearable timeout.
+ */
+class Timeout {
+  /**
+   * Construct a {@link Timeout}.
+   * @param {function} fn - Function to call
+   * @param {number} delay - Delay in milliseconds
+   * @param {Array<*>} args - Optional arguments to be passed to the function
+   */
+  constructor(fn, delay, ...args) {
+    Object.defineProperties(this, {
+      _args: {
+        value: args
+      },
+      _delay: {
+        value: delay
+      },
+      _fn: {
+        value: fn
+      },
+      _timeout: {
+        value: null,
+        writable: true
+      }
+    });
+
+    this._start();
+  }
```

Comment:
Just thinking: it might be nice to have a getter that says whether or not the Timeout is active/live/stopped/etc. something like

```
get isFoo() {
  return this._timeout !== null;
}
```